### ISSUE TITLE
Cop from RuboCop takes precedence over cop with the same name (but different department) from extension and the `require` is done in an inherited file/gem

### ISSUE THREAD

**Issue by rymai** [rubocop/rubocop#5251]
First of all, thanks for this awesome project!

I've encountered a bug where we're [using a gem to define common rules for multiple projects](https://gitlab.com/gitlab-org/gitlab-styles) and [the `require` is done in the gem](https://gitlab.com/gitlab-org/gitlab-styles/blob/bff77b0c7d63b89f458f6d62e27d14db518a29d2/rubocop-default.yml#L2).

Then in a project that uses this gem, [I wanted to disable the `RSpec/FilePath` cop](https://gitlab.com/gitlab-org/gitlab-qa/merge_requests/110/diffs#057477721d04f350346ebfb6554f5c0cd2d8aeaf_27_24).

That wasn't working until I added

```yml
require:
  - rubocop-rspec
```

[in the `.rubocop.yml` of my project](https://gitlab.com/gitlab-org/gitlab-qa/merge_requests/110/diffs#057477721d04f350346ebfb6554f5c0cd2d8aeaf_3_5).

I've set up a reproduction repo here: https://github.com/rymai/reproduce-rubocop-bug

--------

## Expected behavior

1. When you inherit a config from a file or gem,
1. and this config includes a `require`,
1. and the extension defines a cop with the same name (but different department) as a cop from RuboCop itself (e.g. `FilePath`, `rubocop` defines `Rails/FilePath` and `rubocop-rspec` defines `RSpec/FilePath`)
1. Then any rule that acts on the extension cop should be taken in account.

## Actual behavior

The rules that act on the extension cop isn't taken in account.

## Steps to reproduce the problem

See https://github.com/rymai/reproduce-rubocop-bug.

1. `git clone https://github.com/rymai/reproduce-rubocop-bug.git`
1. `cd reproduce-rubocop-bug`
1. `bundle install`
1. `bundle exec rubocop -c .rubocop-inherit-from-gem.yml --show-cops RSpec/FilePath` or `bundle exec rubocop -c .rubocop-inherit-from-file.yml --show-cops RSpec/FilePath`
1. You'll see `Enabled: true` instead of `Enabled: false`

## Work-around

You can work-around this bug by re-requiring the extension in the final RuboCop config file.

## RuboCop version

```
$ rubocop -V
0.52.0 (using Parser 2.4.0.2, running on ruby 2.3.5 x86_64-darwin16)
```

also, `rubocop-rspec (1.21.0)`.

**Comment by bbatsov** [comment]
@bquorning @Darhazer Can you confirm this?

**Comment by rymai** [comment]
> This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Thank you for your contribution and understanding!

After upgrading `rubocop` and `rubocop-rspec` to `0.68.1` and `1.32.0 respectively, it seems that the bug is still occurring:

```
› bundle exec rubocop -c .rubocop-inherit-from-file.yml --show-cops RSpec/FilePath
.rubocop-inherit-from-file.yml: RSpec/FilePath has the wrong namespace - should be Rails
RSpec/FilePath:
  Description: Checks that spec file paths are consistent with the test subject.
  Enabled: true
  CustomTransform:
    RuboCop: rubocop
    RSpec: rspec
  IgnoreMethods: false
  StyleGuide: http://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/FilePath
```

and

```
› bundle exec rubocop -c .rubocop-inherit-from-gem.yml --show-cops RSpec/FilePath 
.rubocop-inherit-from-gem.yml: RSpec/FilePath has the wrong namespace - should be Rails
RSpec/FilePath:
  Description: Checks that spec file paths are consistent with the test subject.
  Enabled: true
  CustomTransform:
    RuboCop: rubocop
    RSpec: rspec
  IgnoreMethods: false
  StyleGuide: http://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/FilePath
```

**Comment by NobodysNightmare** [comment]
Just stumbled upon this one. I was not aware of the workaround using an additional `require` in `.rubocop.yml` directly, so that helped already.

In case this helps (also debugging): Another workaround was to have the configuration inside `.rubocop_todo.yml`, where it would also be correctly recognized.

Can confirm that this is still the case for rubocop 1.39.0.

**Comment by NobodysNightmare** [comment]
Attempting a fix in https://github.com/rubocop/rubocop/pull/11216. Hope this does not break anything else :see_no_evil:
